sockops: rename sock_ops_map to cilium_sock_ops_map

nirmoy opened the pull request:
sock_ops_map was left alone while `cilium cleanup`
`cilium cleanup` use cilium_ prefix for removing
map.

Signed-off-by: Nirmoy Das <ndas@suse.de>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/6665)
<!-- Reviewable:end -->

joestringer reviewed (approved):
If one upgrades from an earlier version of Cilium to a version with this patch, then we'll leave the old path behind. We might want to consider removing the old path on startup (then getting rid of that code after the next release).

EDIT: This would presumably be similar to the code being removed in https://github.com/cilium/cilium/pull/6663.

aanm commented on bpf/sockops/sockops_config.h:
```diff
@@ -15,5 +15,5 @@
  *  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
  */
-#define SOCK_OPS_MAP sock_ops_map
+#define SOCK_OPS_MAP cilium_sock_ops_map
```
@nirmoy can you please add the "Fixes: " commit id that this PR is fixing in the commit message? Something similar to https://github.com/cilium/cilium/commit/0b9c86d271d9ae6de42303565744f604c32b5a5d

nirmoy replied:
added "Fixes" and rebased.

joestringer commented on pkg/sockops/sockops.go:
```diff
@@ -60,7 +60,12 @@ const (
-	sockMap = "sock_ops_map"
+	sockMap = "cilium_sock_ops_map"
+
+	// Cilium version 1.3.0 used this map name, we remove it on
```
I think this should be updated to say 1.4, and I don't think we need to backport this PR to 1.4 just because sockmap is still in some form of tech preview/beta.

nirmoy replied:
@joestringer Thanks, updated and rebased.

joestringer reviewed (changes requested):
Looks good, minor comment update and the PR needs a rebase.

joestringer commented on bpf/sockops/sockops_config.h:
```diff
@@ -16,6 +16,6 @@
  */
 #ifndef SOCK_OPS_MAP
-#define SOCK_OPS_MAP sock_ops_map
+#define SOCK_OPS_MAP cilium_sock_ops_map
```
By the way, in the default config files like this we've started using names like `test_cilium_sock_ops_map`.

nirmoy committed:
sockops: rename sock_ops_map to cilium_sock_ops_map

sock_ops_map was left alone while `cilium cleanup`
`cilium cleanup` use cilium_ prefix for removing
map.

Fixes: 47705aba8d (pkg: bpf: sockops helpers for loading and attaching programs)

Signed-off-by: Nirmoy Das <ndas@suse.de>
